### ISSUE TITLE
Pipelines using parameterized instances

### ISSUE THREAD

**Issue by kcpevey** [pyviz/panel#158]
This is the last line in the Pipeline user guide:  
  
> As you will note the Pipeline stages may be either Parameterized instances or classes, however when working with instances you must ensure that updating the parameters of the class is sufficient to update the current state of the class.
  
I think I need to use Parameterized instances instead of classes in order to get the complexity I need. While I appreciate this warning, I'm wondering... how do I make sure its sufficient? Surely I'll check to make sure its not broken once I build it, but are you suggesting code modifications within my parameterized classes to achieve the updating? Would you anticipate some @param.depends decorators within my parameterized classes?   
  
Basically, the warning makes me worried, but I'm not sure what it is that I should be worried about. @philippjfr  can you clarify?

**Comment by philippjfr** [comment]
Sorry I never replied here.

>Would you anticipate some @param.depends decorators within my parameterized classes?

Yes that's exactly what this warning means. Any state within the class that depends on a parameter should automatically recompute when the parameter is changed. A pipeline can obviously not magically figure out what changes are needed to update the state on the class so the user needs to make sure that changing the parameters triggers any required changes.
